docs/en/Community-Articles/2024-05-09-what-is-object-to-object-mapping/post.md

```diff
@@ -0,0 +1,122 @@
+
+DTOs are used to transfer data of objects from one place to another. We often need to convert our Domain objects to DTOs and DTOs to Domain objects, for this you can use a code like this. 
+
+````csharp
+    public virtual async Task<CustomerDto> CreateAsync(CustomerCreateDto input)
+    {
+
+        var customer = await _customerManager.CreateAsync(
+        input.BirthDay, input.MembershipDate, input.FirstName, input.LastName
+        );
+        CustomerDto customerDto = new CustomerDto
+        {
+            Id = customer.Id,
+            FirstName = input.FirstName,
+            LastName = input.LastName,
+            // ...other
+        };
+        
+        return customerDto;
+    }
+
+````
+This code is repetitive, tedious to write. It violates DRY principles, makes your code very complex and reduces readability.  Instead of using this approach, you can use the automapper which does the matching automatically. Here is a better code:
+
+
+````csharp
+        public virtual async Task<CustomerDto> CreateAsync(CustomerCreateDto input)
+        {
+
+            var customer = await _customerManager.CreateAsync(
+            input.BirthDay, input.MembershipDate, input.FirstName, input.LastName
+            );
+
+            return ObjectMapper.Map<Customer, CustomerDto>(customer);
+        }
+````
+
+# What is AutoMapper?
+
+![Swagger](./images/automapper.png)
+
+Automapper is a .net library that automates object to object mapping. The implementation of Automapper is quite simple. In this chapter I will show you how to use the Automapper library in an ABP-based application. For this reason, I assume that you already have an ABP-based application created. If you have not yet created an ABP-based application, please follow the [Getting Started documentation](https://docs.abp.io/en/abp/latest/Getting-Started-Create-Solution?UI=MVC&DB=EF&Tiered=No).
+
+Create a domain entity similar to this one: 
+
+````csharp
+    public class Customer : FullAuditedAggregateRoot<Guid>
+    {
+        public string? FirstName { get; set; }
+        public string? LastName { get; set; }
+        public DateTime BirthDay { get; set; }
+        public DateTime MembershipDate { get; set; }
+    }
+````
+`Custumer` contains a lot of attributes. we don't want to give all of these attributes to users, so create dto with only the necessary attributes: 
```

Comment:
There is a small typo here (Customer) and also we can improve the paragraph as follows:

```suggestion
`Customer` entity contains some properties (such as `FirstName`, `LastName`, ... and other audited properties coming from the base class - `DeleterId`, `IsDeleted`, `CreationTime` etc. -). Typically, you would not want to show/return all of these properties to end users, at that point, you can create a DTO class and only define the properties that you want to return to the end users.

Let's create the `CustomerGetDto` class in the `*.Application.Contracts` project as follows:
```